### PR TITLE
feat: Added SAM init Connector templates

### PR DESCRIPTION
**Pull request by GavinZZ**
*Issue #, if available:*

*Description of changes:*
Serverless Connector is just released. Implemented some examples for customers to use.
For more detail on connector, https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/managing-permissions-connectors.html

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: nodejs12.x/cookiecutter-aws-sam-step-functions-with-connectors/{{cookiecutter.project_name}}/template.yaml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  # More info about Connector Resource: https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-connector.html
+  SfnToStockCheckerFunctionConnector:
+    Type: AWS::Serverless::Connector 
+    Properties:
+      Source:
+        Id: StockTradingStateMachine
+      Destination:
+        Id: StockCheckerFunction
+      Permissions:
+        - Write
```
Comment (hoffa) [inline nodejs12.x/cookiecutter-aws-sam-step-functions-with-connectors/{{cookiecutter.project_name}}/template.yaml:19]: Consistent with the other resources in the template:

```suggestion
  SfnToStockCheckerFunctionConnector:
    Type: AWS::Serverless::Connector # More info about Connector Resource: https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-connector.html
    Properties:
      Source:
        Id: StockTradingStateMachine
      Destination:
        Id: StockCheckerFunction
      Permissions:
        - Write
```

File: nodejs12.x/cookiecutter-aws-sam-step-functions-with-connectors/{{cookiecutter.project_name}}/template.yaml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+  SfnToStockBuyerFunctionConnector:
+    Type: AWS::Serverless::Connector # More info about Connector Resource: https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-connector.html
```
Comment (hoffa) [inline nodejs12.x/cookiecutter-aws-sam-step-functions-with-connectors/{{cookiecutter.project_name}}/template.yaml:20]: nit: other resources only include this comment for the first instance

**Review by hoffa — APPROVED** [review]

**Review by hawflau — CHANGES_REQUESTED** [review]
Can we add some tests under https://github.com/aws/aws-sam-cli-app-templates/tree/master/tests/integration?

**Comment by GavinZZ** [comment]
> Can we add some tests under https://github.com/aws/aws-sam-cli-app-templates/tree/master/tests/integration?

Oh nice, didn't realize there's tests here. Will do that.

**Comment by GavinZZ** [comment]
Ran the new unit/integ tests with `pytest tests/* -k connector`. All passed.
Screen Shot 2022-10-11 at 11 06 08 AM

**Review by hawflau — APPROVED** [review]

**Review by awood45 — APPROVED** [review]

**Pull request merged**